packages/angular/ssr/test/npm_package/BUILD.bazel

```diff
@@ -0,0 +1,52 @@
+load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
+load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@npm//@bazel/jasmine:index.bzl", "jasmine_node_test")
+load("//tools:defaults.bzl", "ts_library")
+
+ts_library(
+    name = "unit_test_lib",
+    testonly = True,
+    srcs = glob(["**/*.ts"]),
+    deps = [
+        "@npm//@bazel/runfiles",
+    ],
+)
+
+jasmine_node_test(
+    name = "test",
+    srcs = [":unit_test_lib"],
+    data = [
+        "//packages/angular/ssr:npm_package",
+    ],
+)
+
+genrule(
+    name = "critters_license_file",
+    srcs = [
+        "//packages/angular/ssr:npm_package",
+    ],
+    outs = [
+        "THIRD_PARTY_LICENSES.txt",
+    ],
+    cmd = """
+      cp $(location //packages/angular/ssr:npm_package)/third_party/critters/THIRD_PARTY_LICENSES.txt $(location :THIRD_PARTY_LICENSES.txt)
+    """,
+)
+
+diff_test(
+    name = "critters_license_test",
+    failure_message = "\n\nTo accept the new golden file, execute:\nyarn bazel run //packages/angular/ssr/test/npm_package:critters_license_test.accept\n",
+    file1 = ":THIRD_PARTY_LICENSES.txt.golden",
+    file2 = ":critters_license_file",
+)
+
+nodejs_binary(
+    name = "critters_license_test.accept",
+    testonly = True,
+    data = [
+        "THIRD_PARTY_LICENSES.txt.golden",
+        ":unit_test_lib",
+        "//packages/angular/ssr:npm_package",
+    ],
+    entry_point = ":critters-license-test-accept.ts",
```

Comment:
For content I did try to use `copy_file`, but I don’t think there is a way to copy the file to the source tree.

Reply:
Actually I probably managed to find a way to do it, by creating a custom macro

https://github.com/bazelbuild/bazel-skylib/blob/5c071b5006bb9799981d04d74a28bdee2f000d4a/docs/private/stardoc_with_diff_test.bzl#L64